src/app/companyinfo-page/companyinfo-page.component.ts

```diff
@@ -0,0 +1,125 @@
+import { Component, OnInit, AfterViewInit } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { Company } from '../shared/interfaces/company';
+import { Game } from '../shared/interfaces/game';
+import { CompanyService } from '../shared/services/company.service';
+import { GameService } from '../shared/services/game.service'
+import { formSliderParams, exploreScrollParams, ngxSpinnerParams } from '../shared/constants'
+import { NgxSpinnerService } from 'ngx-spinner';
+import { Observable } from 'rxjs';
+
+@Component({
+  selector: 'app-companyinfo-page',
+  templateUrl: './companyinfo-page.component.html',
+  styleUrls: ['./companyinfo-page.component.scss']
+})
+export class CompanyinfoPageComponent implements OnInit, AfterViewInit {
+
+  constructor(private route: ActivatedRoute,
+              private compService: CompanyService,
+              private gameService: GameService,
+              private spinner: NgxSpinnerService ) { }
+
+  private companyId
+  public companyInfo: Company
+  private favourites = []
+
+  //for infinite scroll
+  public notEmptyResp: boolean = true
+  public notScrolly: boolean = true
+  private limit: number = 3
+  private devcurOffset: number = this.limit
+  private pubcurOffset: number = this.limit
+  //
+  public dev_loaded: Observable<Game[]>
+  public pub_loaded: Observable<Game[]>
+  public developed: Game[] = []
+  public published: Game[] = []
+
+  //ui component parameters
+  public formSliderParams = formSliderParams
+  public exploreScrollParams = exploreScrollParams
+  public ngxSpinnerParams = ngxSpinnerParams
+  public selectedTab = 0
+
+  ngOnInit(): void {
+    this.companyId = this.route.snapshot.paramMap.get('id')
+    this.favourites = localStorage.getItem('liked').split(',')
+  }
+
+  ngAfterViewInit(): void {
+    this.compService.getCompanyInfoByID(this.companyId)
+    .then(data => {
+      this.companyInfo = data[0]
+      this.compService.getCompanyLogo([this.companyInfo.logo]).subscribe(data2 => this.companyInfo.logo_url = data2[0].url)
+
+      //filling games lists initial values
+      if(this.companyInfo.developed) {
+        this.dev_loaded = this.gameService.getGamesById(this.companyInfo.developed, this.limit)
+        this.dev_loaded.subscribe(data2 => {
```

Comment:
Check naming convention for RxJS https://angular.io/guide/rx-library#naming-conventions-for-observables